builder/remotecontext/detect.go

```diff
@@ -150,7 +150,17 @@ func readAndParseDockerfile(name string, rc io.Reader) (*parser.Result, error) {
 		}
 		return nil, errors.Wrap(err, "unexpected error reading Dockerfile")
 	}
-	return parser.Parse(br)
+
+	dockerfile, err := parser.Parse(br)
+	if err != nil {
+		return nil, err
+	}
+
+	if dockerfile.AST == nil || dockerfile.AST.StartLine < 0 {
```

Comment:
Actually; wondering if this check should be done inside `parser.Parse()`. Looks like this check will also be needed in this part of the code: https://github.com/moby/moby/blob/dffc96616503936fc282c0e3a3eea8695ee11b67/builder/remotecontext/detect.go#L34-L39

Reply:
I thought that might be the place to put it but it seems like an empty/comment-only dockerfile IS a dockerfile, just not one docker cares to do anything with. However, I'm happy to take the patch there instead if that seems like a better place for it!

Reply:
He just replied; and yes, fixing it upstream would be better 👍

Reply:
Alright! Should I leave the tests here?

Reply:
Yes, let's reuse these test when updating vendoring. In upstream you can add a unit test for the parser package (and another regression integration test for buildkit itself if you want).